jina/serve/runtimes/worker/http_fastapi_app.py

```diff
@@ -82,33 +109,105 @@ async def post(body: input_model, response: Response):
+
+            req = DataRequest()
+            if not docarray_v2:
+                req.data.docs = DocumentArray.from_pydantic_model(body.data)
+            else:
+                req.data.docs = DocList[input_doc_list_model](body.data)
+
+            if body.header is not None:
+                req.header.request_id = body.header.request_id
+
+            req.parameters = body.parameters
+            req.header.exec_endpoint = endpoint_path
+            resp = await caller(req)
+            return EventSourceResponse(resp)
+
+    def add_streaming_get_route(
+        endpoint_path,
+        input_doc_list_model=None,
+    ):
+        from fastapi import Request
+
+        @app.api_route(
+            path=f'/{endpoint_path.strip("/")}',
+            methods=['GET'],
+            summary=f'Streaming Endpoint {endpoint_path}',
+        )
+        async def streaming_get(request: Request):
+            query_params = dict(request.query_params)
+            endpoint = query_params.pop('exec_endpoint')
+            req = DataRequest()
+            req.header.exec_endpoint = endpoint
+            if not docarray_v2:
+                from docarray import Document
+
+                req.data.docs = DocumentArray([Document.from_dict(query_params)])
+            else:
+                req.data.docs = DocumentArray([input_doc_list_model(**query_params)])
+            event_generator = _gen_dict_documents(await caller(req))
+            return EventSourceResponse(event_generator)
+
     for endpoint, input_output_map in request_models_map.items():
         if endpoint != '_jina_dry_run_':
             input_doc_model = input_output_map['input']['model']
             output_doc_model = input_output_map['output']['model']
+            is_generator = input_output_map['is_generator']
 
             endpoint_input_model = pydantic.create_model(
                 f'{endpoint.strip("/")}_input_model',
                 data=(List[input_doc_model], []),
                 parameters=(Optional[Dict], None),
                 header=(Optional[Header], None),
-                __config__=inherit_config(InnerConfig, input_doc_model.__config__)
+                __config__=inherit_config(InnerConfig, input_doc_model.__config__),
             )
 
             endpoint_output_model = pydantic.create_model(
                 f'{endpoint.strip("/")}_output_model',
                 data=(List[output_doc_model], []),
                 parameters=(Optional[Dict], None),
-                __config__=output_doc_model.__config__
+                __config__=output_doc_model.__config__,
             )
 
-            add_route(endpoint,
-                      input_model=endpoint_input_model,
-                      output_model=endpoint_output_model,
-                      input_doc_list_model=input_doc_model,
-                      output_doc_list_model=output_doc_model)
+            if is_generator:
+                add_streaming_post_route(
```

Comment:
I do not understand here, why 2 routes are added?

Reply:
yeah actually the post one is not needed, removed it